src/frontend/sharedComponents/ProjectInviteBottomSheet/index.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import {BottomSheetModal, useBottomSheetModal} from '../BottomSheetModal';
+import {useNavigationState} from '@react-navigation/native';
+import {isEditingScreen} from '../../lib/utils';
+import {
+  useAcceptInvite,
+  usePendingInvites,
+  useRejectInvite,
+} from '../../hooks/server/invites';
+import {useProjectInvitesListener} from '../../hooks/useProjectInvitesListener';
+import {NewInviteBottomSheetContent} from './NewInviteBottomSheetContent';
+import {InviteSuccessBottomSheetContent} from './InviteSuccessBottomSheetContent';
+import {InviteCanceledBottomSheetContent} from './InviteCanceledBottomSheetContent';
+
+export const ProjectInviteBottomSheet = () => {
+  const {sheetRef, isOpen, closeSheet, openSheet} = useBottomSheetModal({
+    openOnMount: false,
+  });
+  const routes = useNavigationState(state => (!state ? [] : state.routes));
+  const index = useNavigationState(state => (!state ? undefined : state.index));
+
+  const isEditScreen = isEditingScreen(routes, index);
+  const invites = usePendingInvites().data.sort(
+    (a, b) => a.receivedAt - b.receivedAt,
+  );
+  const invite = invites[0];
+  const {currentInviteCanceled, resetCacheAndClearCanceled} =
+    useProjectInvitesListener({
+      inviteId: invite?.inviteId,
+      bottomSheetIsOpen: isOpen,
```

Comment:
these are the conditions used to determine if a canceled invite is the invite the user is interacting with.